File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/customer/customer_dto_reader.cpp
#include "customer_dto_reader.h"
#include "customer_dto.h"
#include "../icsv_master_dto.h"
#include "../../util/utilities.h"
#include "../../constants/constants.h"
#include <string>
#include <vector>
#include <cstdlib>
using namespace std;

CustomerDTOReader::CustomerDTOReader() {
	this->file_address = kDefaultCsvFolder + "CUSTOMER_MST.csv";
	AbstractDTOReader::init();
}

CustomerDTOReader::CustomerDTOReader(string &file_address) {
	this->file_address = file_address;
	AbstractDTOReader::init();
}

string CustomerDTOReader::get_file_address() {
	return this->file_address;
}

ICsvMasterDTO *CustomerDTOReader::convert_array_to_dto(vector<string> &value) {
	long customer_id = atol(value[COLUMN_INDEX_CUSTOMER_ID].c_str());
	string name = value[COLUMN_INDEX_NAME];
	string gender = value[COLUMN_INDEX_GENDER];
	string tel = value[COLUMN_INDEX_TEL];
	ICsvMasterDTO *obj = new CustomerDTO(customer_id, name, gender, tel);
	return obj;
}

CustomerDTOReader::~CustomerDTOReader() {

}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/customer/CustomerDTOReader.java
package jp.co.worksap.intern.entities.customer;

import java.io.IOException;

import jp.co.worksap.intern.constants.Constants;
import jp.co.worksap.intern.entities.AbstractDTOReader;

public class CustomerDTOReader extends AbstractDTOReader<CustomerDTO> {
	private static final int COLUMN_INDEX_CUSTOMER_ID = 0;
	private static final int COLUMN_INDEX_NAME = 1;
	private static final int COLUMN_INDEX_GENDER = 2;
	private static final int COLUMN_INDEX_TEL = 3;

	private String fileAddress = Constants.DEFAULT_CSV_FOLDER
			+ "CUSTOMER_MST.csv";

	/**
	 * use default file address
	 * 
	 * @throws IOException
	 */
	public CustomerDTOReader() throws IOException {
		super(CustomerDTOReader.class.getName());
		super.init();
	}

	/**
	 * use customize file address
	 * 
	 * @param fileAddress
	 * @throws IOException
	 */
	public CustomerDTOReader(final String fileAddress) throws IOException {
		super(CustomerDTOReader.class.getName());
		this.fileAddress = fileAddress;
		super.init();
	}

	@Override
	protected String getFileArress() {
		return fileAddress;
	}

	@Override
	protected CustomerDTO convertArrayToDTO(String[] value) throws IOException {

		Long customerId = Long.valueOf(value[COLUMN_INDEX_CUSTOMER_ID]);
		String name = value[COLUMN_INDEX_NAME];
		String gender = value[COLUMN_INDEX_GENDER];
		String tel = value[COLUMN_INDEX_TEL];

		CustomerDTO dto = new CustomerDTO(customerId, name, gender, tel);
		return dto;
	}
}
